### PR TITLE
feat(registry): stop adding last mile ENV Vars into slugs as k8s handles that

### PR DESCRIPTION
**Pull request by helgi**
I am assuming we need to keep the `FROM` portion of the `build` section? We can not just pull the source and push to our own registry with a new name?

ref #13

### CODE REVIEW

**Comment by bacongobbler** [comment]
Perhaps the faster way to do a build would be to just pull, tag then push since that's really all this is doing now.

File: rootfs/registry/dockerclient.py [hunk 1]
```diff
@@ -46,7 +46,6 @@ def publish_release(self, source, config, target, deis_registry):
         self.tag(image, src_name, tag=src_tag)
 
         # build a Docker image that adds a "last-mile" layer of environment
-        config.update({'DEIS_APP': name, 'DEIS_RELEASE': tag})
```
Comment (bacongobbler) [inline rootfs/registry/dockerclient.py]: I think we'll need to inject these into the environment on `create` [here](https://github.com/deis/workflow/blob/15df775432a4d7f87c8d69805e7bab6ce857b838/rootfs/api/models.py#L221) for DEIS_APP and [here](https://github.com/deis/workflow/blob/15df775432a4d7f87c8d69805e7bab6ce857b838/rootfs/api/models.py#L485) for DEIS_RELEASE.

**Comment by smothiki** [comment]
we hit that function if we hit deis pull use case 
https://github.com/deis/workflow/blob/master/rootfs/api/models.py#L880-L881

**Comment by smothiki** [comment]
deis pull apps doesn't fall into buildpack or dockerfile, so we will set the environment variables. Might be pulling up dockerfile template and end up having two environment variables. https://github.com/deis/workflow/blob/master/rootfs/scheduler/__init__.py#L606-L608. Which I'll fix while dealing with issue.
https://github.com/deis/workflow/issues/141

**Comment by bacongobbler** [comment]
@smothiki if that's the case then that's a bug. All three deployment types should have DEIS_APP and DEIS_RELEASE present in their runtime environment. The proposed changes that I suggested in https://github.com/deis/workflow/pull/248#discussion_r49103251 should revert and fix that back to the way things were.

**Comment by smothiki** [comment]
DEIS_RELEASE and DEIS_APP are being set in the replication controller for the rest of the app types.

**Comment by helgi** [comment]
I cut out the build step all together and instead we are doing a pull / push to get it into the deis-registry with all the tagging

File: rootfs/registry/dockerclient.py [hunk 2]
```diff
@@ -45,26 +45,9 @@ def publish_release(self, source, config, target, deis_registry):
...
         self.push("{}/{}".format(self.registry, name), tag)
 
-    def build(self, source, config, repo, tag):
-        """Add a "last-mile" layer of environment config to a Docker image for deis-registry."""
-        check_blacklist(repo)
-        env = ' '.join("{}='{}'".format(
-            k, v.encode('unicode-escape').replace("'", "\\'")) for k, v in config.viewitems())
-        dockerfile = "FROM {}\nENV {}".format(source, env)
-        f = io.BytesIO(dockerfile.encode('utf-8'))
-        target_repo = "{}/{}:{}".format(self.registry, repo, tag)
```
Comment (bacongobbler) [inline rootfs/registry/dockerclient.py]: By removing this line, we are no longer tagging an image called `$REGISTRY_URL/appname:v2`, which will break the push operation on line 49. From what I see, line 46 doesn't really do much other than re-tag the incoming image from `quay.io/mycustomimage:latest` to just `mycustomimage:latest`, stripping the external registry URL. In order to fix this, I think we need to change line 46 to something like:

```
self.tag(image, "{}/{}".format(self.registry, name), tag=tag)
```

That way the image should be properly bagged and tagged for shipping to the registry. I'll verify my thoughts manually tomorrow, but if you have more insight on how this works then I'm prepared to be wrong :)

Comment (helgi) [inline rootfs/registry/dockerclient.py]: Doneeeeee and it works on `deis pull`

**Comment by bacongobbler** [comment]
:shipit:

**Comment by helgi** [comment]
I tried to use https://github.com/docker/docker-py/blob/master/docker/api/image.py#L45 to see if that could simplify anything but didn't have any luck (yet). Doesn't seem to do the kind of tagging we need nor does it appear to block but I may have just used it wrong :D

**Pull request merged**